src/NUnitEngine/nunit.engine.core/DotNetHelper.cs

```diff
@@ -36,18 +38,43 @@ public static class DotNet
 
         public class RuntimeInfo
         {
-            public string Name;
-            public Version Version;
-            public string Path;
+            public string Name { get; }
+            public string PackageVersion { get; }
+            public string Path { get; }
 
-            public RuntimeInfo(string name, string version, string path)
-                : this(name, new Version(version), path) { }
+            public bool IsPreRelease { get; }
+            public Version Version { get; }
+            public string PreReleaseSuffix { get; }
 
-            public RuntimeInfo(string name, Version version, string path)
+
+            public RuntimeInfo(string name, string packageVersion, string path)
             {
                 Name = name;
-                Version = version;
+                PackageVersion = packageVersion;
                 Path = path;
+
+                int dash = PackageVersion.IndexOf('-');
+                IsPreRelease = dash > 0;
+
+                if (IsPreRelease)
+                {
+                    Version = new Version(PackageVersion.Substring(0, dash));
+                    PreReleaseSuffix = PackageVersion.Substring(dash + 1);
+                }
+                else
+                    Version = new Version(packageVersion);
```

Comment:
Inconsistent. Uses `PackageVersion` property in `if` and `packageVersion` parameter in `else`.
More important `PreReleaseSuffix` will be `null` in the `else` case.